### PR TITLE
Update go.mod with golang and dependencies

### PR DESCRIPTION
**Pull request by Rajalakshmi-Girish**
Steps followed:
1. Changed golang version in `go.mod`
2. Did `go mod tidy`
3. Did `go get -u sigs.k8s.io/kubetest2@master`
4. Plugin gave below error due to upstream change in `kubetest2` https://github.com/kubernetes-sigs/kubetest2/commit/7f5e6db62551cac73f662821645c36389376174e
```
# github.com/ppc64le-cloud/kubetest2-plugins/kubetest2-tf/deployer
kubetest2-tf/deployer/deployer.go:92:36: opts.ArtifactsDir undefined (type types.Options has no field or method ArtifactsDir)
```
5. Hence changed below:
```
[root@rajalakshmi-workspace1 kubetest2-plugins]# git diff kubetest2-tf/deployer/deployer.go
diff --git a/kubetest2-tf/deployer/deployer.go b/kubetest2-tf/deployer/deployer.go
index 339aefc..de0685d 100644
--- a/kubetest2-tf/deployer/deployer.go
+++ b/kubetest2-tf/deployer/deployer.go
@@ -89,7 +89,7 @@ var (
 func New(opts types.Options) (types.Deployer, *pflag.FlagSet) {
        d := &deployer{
                commonOptions: opts,
-               logsDir:       filepath.Join(opts.ArtifactsDir(), "logs"),
+               logsDir:       filepath.Join(opts.RunDir(), "logs"),
        }
        return d, bindFlags(d)
 }
[root@rajalakshmi-workspace1 kubetest2-plugins]#
```
6. Did `go mod tidy` again

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,19 +1,45 @@
...
+	gopkg.in/inf.v0 v0.9.1 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
+	k8s.io/apimachinery v0.19.1 // indirect
+	k8s.io/klog v1.0.0 // indirect
+	k8s.io/utils v0.0.0-20210111153108-fddb29f9d009 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.0.1 // indirect
+	sigs.k8s.io/yaml v1.2.0 // indirect
 )
 
 replace k8s.io/client-go => k8s.io/client-go v0.19.1
```
Comment (mkumatag) [inline go.mod:45]: can you update this to latest version as well?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (Rajalakshmi-Girish) [reply go.mod:45]: @mkumatag did you mean the version of k8s.io/client-go?

Comment (Rajalakshmi-Girish) [reply go.mod:45]: @mkumatag did you mean the version of k8s.io/client-go?

Comment (mkumatag) [reply go.mod:45]: yes

Comment (Rajalakshmi-Girish) [reply go.mod]: @mkumatag PTAL.
I changed the k8s.io.client version to the latest 0.24.0 and tested the `kubetest2-tf` binary generated with new changes.

**Review by mkumatag — APPROVED** [review]
/lgtm
/approve

**Pull request merged**